src/main/app/controller/courts/CasesHeardController.ts

```diff
@@ -0,0 +1,65 @@
+import autobind from 'autobind-decorator';
+import {AuthedRequest} from '../../../types/AuthedRequest';
+import {Response} from 'express';
+import {AreaOfLaw} from '../../../types/AreaOfLaw';
+import {Error} from '../../../types/Error';
+import {CasesHeardPageData} from '../../../types/CasesHeardPageData';
+import {CSRF} from '../../../modules/csrf';
+import {AxiosError} from 'axios';
+
+@autobind
+export class CasesHeardController {
+
+  getAreasOfLawErrorMsg = 'A problem occurred when retrieving the areas of law. ';
+  getCourtAreasOfLawErrorMsg = 'A problem occurred when retrieving the court areas of law. ';
+  putCourtAreasOfLawErrorMsg = 'A problem occurred when updating the court areas of law. ';
+
+  public async get(
+    req: AuthedRequest,
+    res: Response,
+    error = '',
+    updated = false,
+    allAreasOfLaw: AreaOfLaw[] = null,
+    courtAreasOfLaw: AreaOfLaw[]): Promise<void> {
+    const slug: string = req.params.slug as string;
+    if (!allAreasOfLaw) {
+      await req.scope.cradle.api.getAllAreasOfLaw()
+        .then((value: AreaOfLaw[]) => allAreasOfLaw = value)
+        .catch(() => error += this.getAreasOfLawErrorMsg);
+    }
+    if (!courtAreasOfLaw) {
+      await req.scope.cradle.api.getCourtAreasOfLaw(slug)
+        .then((value: AreaOfLaw[]) => courtAreasOfLaw = value)
+        .catch(() => error += this.getCourtAreasOfLawErrorMsg);
+    }
+    const errors: Error[] = [];
+    if (error) {
+      errors.push({text: error});
```

Comment:
Rather than concatenating the error strings and then adding them to the array as a single element, I think it would be better just to push each error message to the array when the error is caught on lines 28 and 33. There's an example of this in the OpeningTimeController class.